scripts/zmfilter.pl.in

```diff
@@ -475,7 +475,8 @@ sub uploadArchFile {
   }
 
   my $archFile = $event->{MonitorName}.'-'.$event->{Id};
-  my $archImagePath = $event->Path()
+  #my $archImagePath = $event->Path()
+  my $archImagePath = join('/', $Config{ZM_DIR_EVENTS},$event->{MonitorId},$event->{Id})
```

Comment:
This is wrong.  The reason to use $Event->Path() is because the event object knows how to construct the path, depending on the storage scheme. This is why we are passing in the object instead of the db row.
ZM_DIR_EVENTS is deprecated.

Reply:
When I run the original, the constructions failed with an `unblessed reference` error, reproducibly. I also saw that the Event object knows how to construct its own path, but evidently it does not work. Moving the same logic here _did_ allow the upload to run (after the `sftOptions` object was fixed).

Yet, if that isn't the cause, then what am I masking?